test/unit-test/FreeRTOS_DNS/FreeRTOS_DNS_utest.c

```diff
@@ -0,0 +1,4263 @@
+/* Include Unity header */
+#include "unity.h"
+
+/* Include standard libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "mock_list.h"
+/* This must come after list.h is included. */
+#include "mock_list_macros.h"
+#include "mock_task.h"
+#include "mock_semphr.h"
+#include "mock_FreeRTOS_IP.h"
+#include "mock_FreeRTOS_Sockets.h"
+#include "mock_FreeRTOS_UDP_IP.h"
+#include "mock_FreeRTOS_DHCP.h"
+#include "mock_FreeRTOS_IP_Private.h"
+#include "mock_NetworkBufferManagement.h"
+
+#include "mock_portable_functions.h"
+#include "mock_FreeRTOS_DNS_mock.h"
+
+#include "FreeRTOS_DNS.h"
+
+#include "FreeRTOS_DNS_stubs.c"
+#include "catch_assert.h"
+
+#include "FreeRTOSIPConfig.h"
+
+extern DNSCacheRow_t xDNSCache[ ipconfigDNS_CACHE_ENTRIES ];
+
+DNSCacheRow_t xTestDNSCache[ ipconfigDNS_CACHE_ENTRIES ];
+
+List_t xCallbackList;
+
+/* Setup the environment for the ParseDNSReply function call. */
```

Comment:
I find this exceedingly difficult to follow.  It's not apparent to me that this achieves the degree of isolation needed for reasonable unit tests.